dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBNoSQL/AzureCosmosDBNoSQLVectorStoreCollectionQueryBuilder.cs

```diff
@@ -0,0 +1,195 @@
+        var vectorDistanceArgument = $"VectorDistance({tableVariableName}.{vectorPropertyName}, {VectorVariableName})";
+        var vectorDistanceArgumentWithAlias = $"{vectorDistanceArgument} AS {scorePropertyName}";
+
+        var selectClauseArguments = string.Join(SelectClauseDelimiter, [.. fieldsArgument, vectorDistanceArgumentWithAlias]);
+
+        var filter = BuildSearchFilter(searchOptions.Filter, storagePropertyNames);
+
+        var filterQueryParameters = filter?.QueryParameters;
+        var filterWhereClauseArguments = filter?.WhereClauseArguments;
+
+        var whereClause = filterWhereClauseArguments is { Count: > 0 } ?
+            $"WHERE {string.Join(AndConditionDelimiter, filterWhereClauseArguments)}" :
+            string.Empty;
+
+        var query =
+            $"SELECT {selectClauseArguments} " +
+            $"FROM {tableVariableName} " +
+            $"{whereClause} " +
+            $"ORDER BY {vectorDistanceArgument} " +
+            $"OFFSET {OffsetVariableName} LIMIT {LimitVariableName} ";
+
+        var queryDefinition = new QueryDefinition(query);
+
+        queryDefinition.WithParameter(VectorVariableName, vector);
+        queryDefinition.WithParameter(OffsetVariableName, searchOptions.Offset);
+        queryDefinition.WithParameter(LimitVariableName, searchOptions.Limit);
+
+        if (filterQueryParameters is { Count: > 0 })
+        {
+            foreach (var queryParameter in filterQueryParameters)
+            {
+                queryDefinition.WithParameter(queryParameter.Key, queryParameter.Value);
+            }
+        }
+
+        return queryDefinition;
+    }
+
+    /// <summary>
+    /// Builds <see cref="QueryDefinition"/> to get items from Azure CosmosDB NoSQL.
+    /// </summary>
+    public static QueryDefinition BuildSelectQuery(
+        string keyStoragePropertyName,
+        string partitionKeyStoragePropertyName,
+        List<AzureCosmosDBNoSQLCompositeKey> keys,
+        List<string> fields)
+    {
+        Verify.True(keys.Count > 0, "At least one key should be provided.", nameof(keys));
+
+        const string RecordKeyVariableName = "@rk";
+        const string PartitionKeyVariableName = "@pk";
+
+        var tableVariableName = AzureCosmosDBNoSQLConstants.TableQueryVariableName;
+
+        var selectClauseArguments = string.Join(SelectClauseDelimiter,
+            fields.Select(field => $"{tableVariableName}.{field}"));
+
+        var whereClauseArguments = string.Join(OrConditionDelimiter,
+            keys.Select((key, index) =>
+                $"({tableVariableName}.{keyStoragePropertyName} = {RecordKeyVariableName}{index} {AndConditionDelimiter} " +
+                $"{tableVariableName}.{partitionKeyStoragePropertyName} = {PartitionKeyVariableName}{index})"));
+
+        var query =
+            $"SELECT {selectClauseArguments} " +
+            $"FROM {tableVariableName} " +
+            $"WHERE {whereClauseArguments} ";
```

Comment:
Are SQL injection attacks a concern?

Reply:
Anything that can come from end users is injected into query using query parameters - `QueryDefinition.WithParameter()`, which should prevent from SQL injection attacks.